py3status/modules/wwan_status.py

```diff
@@ -0,0 +1,178 @@
+                    bytesize=serial.EIGHTBITS)
+                if ser.isOpen():
+                    ser.close()
+                ser.open()
+                ser.write((query + "\r").encode())
+                print("Issued query to " + self.modem)
+                sleep(self.modem_timeout)
+                n = ser.inWaiting()
+                modem_response = ser.read(n)
+                ser.close()
+            except:
+                # This will happen...
+                # 1) in the short timespan between the creation of the device node
+                # and udev changing the permissions. If this message persists,
+                # double check if you are using the proper device file
+                # 2) if/when you unplug the device
+                PermissionError
+                print("Permission error")
+                response['full_text'] = self.format_error.format(
+                    error="no access to " + self.modem)
+                response['color'] = i3s_config['color_bad']
+                return response
+            # Dissect response
+            for line in modem_response.decode("utf-8").split('\n'):
+                print(line)
+                if line.startswith(target_line):
+                    # Determine IP once the modem responds
+                    ip = self.get_ip(self.interface)
+                    if not ip:
+                        ip = "no ip"
+                    modem_answer = line.split(',')
+                    netgen = len(modem_answer[-2]) + 1
+                    netmode = modem_answer[-1].rstrip()[1:-1]
+                    if netmode == "NO SERVICE":
+                        response['full_text'] = self.format_no_service.format(
+                            status=netmode,
+                            ip=ip)
+                        response['color'] = i3s_config['color_bad']
+                    else:
+                        response['full_text'] = self.format_up.format(
+                            status=netmode,
+                            netgen=str(netgen) + "G",
+                            ip=ip)
+                        if netgen <= degraded_netgen:
+                            response['color'] = i3s_config['color_degraded']
+                        else:
+                            response['color'] = i3s_config['color_good']
+                elif line.startswith("COMMAND NOT SUPPORT") or line.startswith(
+                        "ERROR"):
+                    response['color'] = i3s_config['color_bad']
+                    response['full_text'] = self.format_error.format(
+                        error="unsupported modem")
+                else:
+                    # Outputs can be multiline, so just try the next one
+                    pass
+        else:
+            print(self.modem + " not found")
+            response['color'] = i3s_config['color_bad']
+            response['full_text'] = self.format_down
+        return response
+
+    def get_ip(self, interface):
```

Comment:
this should be a private method `_get_ip` else it will be executed by py3status directly too